pkgs/tools/admin/balena-cli/default.nix

```diff
@@ -21,27 +21,63 @@ let
     aarch64-darwin = "1ihxyf35px3s6q2yk4p3dy03rcj93hy96bj3pxqlv0rp05gnsf02";
     x86_64-windows = "104hc3qvs04l2hmjmp0bcjr5g5scp4frhprk1fpszziqhdmhwa40";
   }.${system} or throwSystem;
-in
-stdenv.mkDerivation rec {
-  pname = "balena-cli";
-  version = "15.1.1";
 
+  version = "15.1.1";
   src = fetchzip {
     url = "https://github.com/balena-io/balena-cli/releases/download/v${version}/balena-cli-v${version}-${plat}-standalone.zip";
     inherit sha256;
   };
+in
+stdenv.mkDerivation rec {
+  pname = "balena-cli";
+  inherit version src;
 
   installPhase = ''
-    runHook preInstall
```

Comment:
Why are these removed?

Reply:
runHook aren't added to post/pre phases. But for the main phase. (installPhase, checkPhase, buildPhase, etc.) they are added, which triggers pre/post phases. Because you are replacing the installPhase, the hooks have to be manually added.

I think it is possible to improve on this. By not relying on main phase to run hooks. But then, something else would have to be used to mess with hooks. But that would be a thread on it's own.

Reply:
Ah, ok makes sense. Really appreciate the clarification!